### DISCUSSION TITLE
How to prevent auto visual mode when mouse select

### DISCUSSION THREAD

**Discussion by teal024** [JetBrains/ideavim#637] [Q&A]
Ideavim enters visual mode on mouse select automaticly. In vim/neovim/vscode vim plugin, you can simply adjust it in your vimrc. Yet in ideavim, commands like "set mouse-=a" simply won't work. Solutions like "set selectmode=mouse" will change it to vim-select mode. I wonder if there being other solutions that keeps the original mode when you use mouse to select text.

**Comment by AlexPl292** [comment]
Hi, I'm not sure how do you expect this to work? If you turn off the mouse in vim using `set mouse-=a`, you turn off the mouse in vim application. The selection that is created with that is a part of the terminal application.
As I know, in vim there is no mode that has a selection, but not visual or select mode.

As I understand, this is also a reply from you, so I'll answer it here. https://youtrack.jetbrains.com/issue/VIM-2446

**Reply by teal024** [reply]
Sorry about not describing it clearly, what I said is related to https://stackoverflow.com/questions/32966758/disabling-mouse-support-on-ideavim.  The editor always goes into visual mode when I accidentally drag with the mouse. I hope to keep the original mode that I use.

**Reply by AlexPl292** [reply]
Okay, I see the point. We don't any close plans to support this option, but you can track this issue: https://youtrack.jetbrains.com/issue/VIM-2037/Support-mouse-option
